### PR TITLE
Target envvars

### PR DESCRIPTION
**Pull request by joe-kimmel-vmw**

### CODE REVIEW

File: cmd/lifecycle/builder.go [hunk 1]
```diff
@@ -67,10 +67,14 @@ func (b *buildCmd) Exec() error {
 	if err = verifyBuildpackApis(group); err != nil {
 		return err
 	}
-	return b.build(group, plan)
+	amd, err := platform.ReadAnalyzed(b.AnalyzedPath, cmd.DefaultLogger) // TODO: how to verify that AnalyzedPath is set correctly...?
```
Comment (natalieparellano) [inline cmd/lifecycle/builder.go:70]: I think we also need to add 
```
	case b.PlatformAPI.AtLeast("0.12"):
		cli.FlagAnalyzedPath(&b.AnalyzedPath)
```
around line 24

File: buildpack/detect.go [hunk 2]
```diff
@@ -23,6 +23,10 @@ const (
 	EnvExtensionDir = "CNB_EXTENSION_DIR"
 	// EnvPlatformDir is the absolute path of the platform directory (read-only); a single copy is provided for all buildpacks
 	EnvPlatformDir = "CNB_PLATFORM_DIR"
+	// EnvTargetOS is the target os
+	EnvTargetOS = "CNB_TARGET_OS"
+	// EnvTargetArch is the target architecture.
+	EnvTargetArch = "CNB_TARGET_ARCH"
```
Comment (natalieparellano) [inline buildpack/detect.go:29]: Are these used anywhere?

File: buildpack/generate.go [hunk 3]
```diff
@@ -97,6 +102,10 @@ func runGenerateCmd(d ExtDescriptor, extOutputDir, planPath string, inputs Gener
 		EnvOutputDir+"="+extOutputDir,
 		EnvPlatformDir+"="+inputs.PlatformDir,
 	)
+	if api.MustParse(d.WithAPI).AtLeast("0.12") {
+		// todo - nil check?
+		cmd.Env = append(cmd.Env, "") // TODO platform.EnvVarsFor(&(inputs.Target))...)
+	}
```
Comment (natalieparellano) [inline buildpack/generate.go:108]: I don't think we need to do anything in the buildpack package, if the vars are just handed to us with the rest of the env

File: generator_test.go [hunk 4]
```diff
@@ -248,7 +250,47 @@ func testGenerator(t *testing.T, when spec.G, it spec.S) {
 			_, err := generator.Generate()
 			h.AssertNil(t, err)
 		})
-
+		it.Focus("passes through CNB_TARGET environment variables", func() {
```
Comment (natalieparellano) [inline generator_test.go:253]: Focus :)

File: generator.go [hunk 5]
```diff
@@ -134,6 +139,9 @@ func (g *Generator) Generate() (GenerateResult, error) {
 		g.Logger.Debug("Finding plan")
 		inputs.Plan = filteredPlan.Find(buildpack.KindExtension, ext.ID)
 
+		if g.AnalyzedMD.RunImage != nil && g.AnalyzedMD.RunImage.TargetMetadata != nil && g.PlatformAPI.AtLeast("0.12") {
+			inputs.Env = env.NewBuildEnv(append(inputs.Env.List(), platform.EnvVarsFor(g.AnalyzedMD.RunImage.TargetMetadata)...))
+		}
```
Comment (natalieparellano) [inline generator.go:144]: I think we also need something like this in detector.go

File: buildpack/generate.go [hunk 6]
```diff
@@ -7,6 +7,8 @@ import (
 	"os/exec"
 	"path/filepath"
 
+	// "github.com/buildpacks/lifecycle/platform"
+
```
Comment (natalieparellano) [inline buildpack/generate.go:11]:
```suggestion
```

File: buildpack/generate.go [hunk 7]
```diff
@@ -27,6 +29,7 @@ type GenerateInputs struct {
 	Env            BuildEnv
 	Out, Err       io.Writer
 	Plan           Plan
+	// Target         platform.TargetMetadata
```
Comment (natalieparellano) [inline buildpack/generate.go:32]:
```suggestion
```

File: cmd/lifecycle/builder.go [hunk 8]
```diff
@@ -22,14 +22,12 @@ type buildCmd struct {
 // DefineFlags defines the flags that are considered valid and reads their values (if provided).
 func (b *buildCmd) DefineFlags() {
 	switch {
+	case b.PlatformAPI.AtLeast("0.12"):
+		cli.FlagAnalyzedPath(&b.AnalyzedPath)
+		fallthrough
```
Comment (natalieparellano) [inline cmd/lifecycle/builder.go:27]: Nice improvement, thanks :)

File: cmd/lifecycle/builder.go [hunk 9]
```diff
@@ -67,10 +65,14 @@ func (b *buildCmd) Exec() error {
 	if err = verifyBuildpackApis(group); err != nil {
 		return err
 	}
-	return b.build(group, plan)
+	amd, err := platform.ReadAnalyzed(b.AnalyzedPath, cmd.DefaultLogger) // TODO: how to verify that AnalyzedPath is set correctly...?
```
Comment (natalieparellano) [inline cmd/lifecycle/builder.go:68]:
```suggestion
	amd, err := platform.ReadAnalyzed(b.AnalyzedPath, cmd.DefaultLogger)
```

**Review by natalieparellano — COMMENTED** [review]
Looks good to me! A couple nits about outdated comments

**Comment by joe-kimmel-vmw** [comment]
Thanks @natalieparellano -  but also, gross - my apologies for missing all those commented lines.

**Review by natalieparellano — APPROVED** [review]
Looks good to me :)

**Pull request merged**